### PR TITLE
Bug 1427305 - only render source URLs starting with 'https://'

### PR DESCRIPTION
**Pull request by djmitche**

### CODE REVIEW

File: src/views/UnifiedInspector/TaskDetails.js [hunk 1]
```diff
@@ -122,13 +126,13 @@ export default class TaskDetails extends React.PureComponent {
...
               <td>
-                <a href={metadata.source}>
-                  <span>
-                    {metadata.source.length > 90
-                      ? `...${metadata.source.substr(8 - 90)}`
-                      : metadata.source}
-                  </span>
-                </a>
+                {metadata.source.startsWith('https://') ? (
+                  <a href={metadata.source}>
```
Comment (eliperelman) [inline src/views/UnifiedInspector/TaskDetails.js]: I just realized we should probably put `rel="noopener noreferrer"` here.

Comment (djmitche) [reply src/views/UnifiedInspector/TaskDetails.js]: What's the advantage?  I don't think there's any private information in the referrer URL..

Comment (eliperelman) [reply src/views/UnifiedInspector/TaskDetails.js]: It's not about private information, it's more about giving the opening page a JS context into tools, where credentials could be accessed, i.e. tabnabbing.

See also https://mathiasbynens.github.io/rel-noopener/

Comment (djmitche) [reply src/views/UnifiedInspector/TaskDetails.js]: Maybe we should just avoid ever linking?  Anyway, this sounds like a good fix.

Comment (eliperelman) [reply src/views/UnifiedInspector/TaskDetails.js]: I mean, that's technically possible, but:

```
src/components/Navigation/index.js:                      rel="noopener noreferrer">
src/components/ArtifactList/index.js:            rel="noopener noreferrer"
src/components/ArtifactList/index.js:            <a href={url} target="_blank" rel="noopener noreferrer">
src/views/AwsProvisioner/WorkerTypeResources.js:      <a href={`${awsUrl}${qs}`} target="_blank" rel="noopener noreferrer">
src/views/AwsProvisioner/WorkerTypeResources.js:      <a href={`${awsUrl}${qs}`} target="_blank" rel="noopener noreferrer">
src/views/AwsProvisioner/WorkerTypeResources.js:      <a href={`${awsUrl}${qs}`} target="_blank" rel="noopener noreferrer">
src/views/Home/index.js:                    rel="noopener noreferrer">
src/views/WorkerManager/WorkerTable.js:          <a target="_blank" rel="noopener noreferrer" href={source}>
src/views/HooksManager/HookEditor.js:                rel="noopener noreferrer">
src/views/HooksManager/HookEditor.js:                rel="noopener noreferrer">
src/views/QuickStart/index.js:                  rel="noopener noreferrer">
src/views/QuickStart/index.js:                rel="noopener noreferrer">
src/views/TaskCreator/TaskCreator.js:            rel="noopener noreferrer">
src/views/UnifiedInspector/TaskDetails.js:                  rel="noopener noreferrer">
src/views/UnifiedInspector/LogView.js:            rel="noopener noreferrer"
```

Comment (djmitche) [reply src/views/UnifiedInspector/TaskDetails.js]: I was just expressing existential despair, not being serious.

Comment (eliperelman) [reply src/views/UnifiedInspector/TaskDetails.js]: 🤣

**Pull request merged**